api/rocdecode.h

```diff
@@ -150,7 +150,6 @@ typedef struct _RocdecDecodeCaps {
     uint32_t bit_depth_minus_8;            /**< IN: The Value "BitDepth minus 8" */
     uint32_t reserved_1[3];                /**< Reserved for future use - set to zero */
     uint8_t is_supported;                  /**< OUT: 1 if codec supported, 0 if not supported */
-    uint8_t num_decoders;                  /**< OUT: Number of Decoders that can support IN params */
```

Comment:
Driver does not provide this info through VAAPI. We don't want to hard code. My suggestion is to remove it for now. 

Regarding maximizing total decode throughput, I don't think we can do much at app level other than multi-threading since we can not do any load balancing. We can not even choose which decoder instance to launch the decode session on. At app level, all we can do is to launch as many threads as we need, regardless the HW decoder number.

Lets discuss in the meeting.

Reply:
The VA-API does not provide a way to determine the number of VCN instances, but the amd_gpu DRM has an API `amdgpu_query_hw_ip_count(device_handle, AMDGPU_HW_IP_VCN_DEC, &count);` that can be used to query the number of decoders.

My point is that the application, by querying the `num_decoders`, can accurately determine how many threads should be launched to achieve maximum throughput. For example, if `num_decoders = 4`, then launching 4 threads should be sufficient to achieve the potential maximum throughput.

Reply:
Number of instance is only one of the factors of the total throughput, the single engine max decode throughput is also critical, plus the load balancing policy. So with only num_decoders, we actually don't know if the total max throughput is reached or not. For example, if one decoder instance can decode 1080 at 1000 fps, we can either send two 30fps 1080 streams to one decoder, or send them to two decoders. The result will be the same (not frame drops). In use case of, say training, we want to decode as fast as possible. We will want to do multi-threading as much as possible, regardless the underlying HW instance number.

There is use case of number of decoder instance on another platform which I can explain in the meeting.

Reply:
Yes ticket SWDEV-417810 shows a way to probe check VCN/JPEG engine numbers. But we probably want to avoid low level private API calls if we don't use the result at all.